src/main.cpp

```diff
@@ -1505,14 +1483,39 @@ bool CheckTransactionWithoutProofVerification(const CTransaction& tx, CValidatio
         }
     }
 
-    auto valueBalanceOrchard = orchard_bundle.GetValueBalance();
+    // nSpendsSapling, nOutputsSapling, and nActionsOrchard MUST all be less than 2^16
+    size_t max_elements = (1 << 16) - 1;
+    if (tx.vShieldedSpend.size() > max_elements) {
+        return state.DoS(
+            dosLevelPotentiallyRelaxing,
+            error("ContextualCheckTransaction(): 2^16 or more Sapling spends"),
```

Comment:
```suggestion
            error("CheckTransactionWithoutProofVerification(): 2^16 or more Sapling spends"),
```

Reply:
I'd rather we change this to `CheckTransaction`. The only reason `*WithoutProofVerification` exists is a Sprout artifact that I want to refactor away, after which we could undo the split.